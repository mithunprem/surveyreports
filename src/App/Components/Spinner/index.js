import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import { Spinner as LoadingSpinner } from 'reactstrap';

const Spinner = ({ spinnerText }) => {
  return (
    <Fragment>
      <div className="m-4">
        { spinnerText } <LoadingSpinner type="grow" color="dark" />
      </div>
    </Fragment>
  );
}

Spinner.propTypes = {
  spinnerText: PropTypes.string
}

export default Spinner;
