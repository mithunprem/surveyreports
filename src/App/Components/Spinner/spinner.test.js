import React from 'react';
import { create } from 'react-test-renderer';
import Spinner from '../../Components/Spinner';

describe("Spinner", () => {
  test("should render without throwing an error", () => {
    const loadingSpinner = create(<Spinner spinnerText="Test" />).toJSON();
    expect(loadingSpinner).toMatchSnapshot();
  });
});
