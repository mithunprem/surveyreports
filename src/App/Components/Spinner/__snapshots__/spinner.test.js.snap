// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner should render without throwing an error 1`] = `
<div
  className="m-4"
>
  Test
   
  <div
    className="spinner-grow text-dark"
    role="status"
  >
    <span
      className="sr-only"
    >
      Loading...
    </span>
  </div>
</div>
`;
